Makefile.am

```diff
@@ -274,8 +274,23 @@ cc_proxy_extra_dist =			\
 
 CHECK_DEPS += check-proxy
 
+define CONTENT
+package main
+import "fmt"
+func main() {
+	fmt.Printf("Executing test program")
+}
+endef
+export CONTENT
+
 check-proxy:
-	go test -v -race -timeout 2s $(srcdir)/proxy
+	echo "$$CONTENT">./racetest.go
+	if ! go build -race ./racetest.go; then \
+		go test -v -race -timeout 2s $(srcdir)/proxy; \
```

Comment:
It would be better to create a variable for the extra argument, rather than having two almost identical `go test` calls:

```
go_test_extra_args=
  :
# set if "go build" supports -race
go_test_extra_args+= -race
  :
go test -v $(go_test_extra_args) -timeout 2s $(srcdir)/proxy
```

That would make the code clearer, cleaner and easier to maintain.